### PR TITLE
implement configGetParameter and configSetParameter

### PR DESCRIPTION
**Pull request by ariovistus**
at least for the ParamEnums I know what to do with

### CODE REVIEW

File: gen/MotController_data.yml [hunk 1]
```diff
@@ -95,6 +95,9 @@ c_MotController_Create1:
       'limit_switch_source_rev': LimitSwitchSource.Deactivated,
       'limit_switch_follow_for': deviceId,
       'limit_switch_follow_rev': deviceId,
+
+      'clear_pos_on_limit_fwd': 0,
```
Comment (auscompgeek) [inline gen/MotController_data.yml]: Better to store whether these are set using a bool.

File: gen/MotController_data.yml [hunk 2]
```diff
@@ -1125,7 +1128,48 @@ c_MotController_ConfigGetCustomParam:
...
+    elif param == ParamEnum.eNeutralDeadband:
+        self.hal_data['neutral_deadband'] = value
+    elif param == ParamEnum.ePeakPosOutput:
+        self.hal_data['peak_fwd_output'] = value
+    elif param == ParamEnum.eNominalPosOutput:
+        self.hal_data['nom_fwd_output'] = value
+    elif param == ParamEnum.ePeakNegOutput:
+        self.hal_data['peak_rev_output'] = value
+    elif param == ParamEnum.eNominalNegOutput:
+        self.hal_data['nom_rev_output'] = value
```
Comment (auscompgeek) [inline gen/MotController_data.yml]: I suspect the PID param values sent are different to what the config API accepts.

**Comment by auscompgeek** [comment]
Darn, I'm slow.

**Comment by virtuald** [comment]
I suspect you actually reviewed it whereas I just glanced at it :+1:

File: gen/MotController_data.yml (hunk 2 above)
Comment (auscompgeek) [reply gen/MotController_data.yml]: … and this mobile client is broken and attached these comments to the wrong line >_>

File: gen/MotController_data.yml (hunk 1 above)
Comment (ariovistus) [reply gen/MotController_data.yml]: configGetParameter and configSetParameter are autogenned to require floats, which makes that awkward

Comment (auscompgeek) [reply gen/MotController_data.yml]: You should probably fix the tests to actually test this then.

**Pull request merged**